tokio-trace-serde/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+use std::fmt;
+
+use serde::{
+    Serialize,
+    ser::{
+        SerializeMap,
+        SerializeStruct,
+        Serializer,
+    },
+};
+
+use tokio_trace::{
+    event::Event,
+    field::{Field, Visit},
+    metadata::Metadata,
+    span::{Attributes, Record},
+};
+
+#[derive(Debug)]
+pub struct FieldVisitor(pub (crate) Field);
+
+impl Serialize for FieldVisitor {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: Serializer {
+        let mut state = serializer.serialize_struct("Field", 1)?;
+        state.serialize_field("name", self.0.name())?;
+        state.end()
+    }
+}
+
+#[derive(Debug)]
+pub struct SerializeMetadata<'a>(pub (crate) &'a Metadata<'a>);
+
+impl<'a> Serialize for SerializeMetadata<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: Serializer {
+        let mut state = serializer.serialize_struct("Metadata", 2)?;
+        state.serialize_field("name", self.0.name)?;
+        state.serialize_field("target", self.0.target)?;
+        state.end()
+    }
+}
+
+/// A Serde visitor to pull `Event` data out of a serialized stream
+#[derive(Debug)]
+pub struct SerializeEvent<'a>(pub (crate) Event<'a>);
+
+impl<'a> Serialize for SerializeEvent<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: Serializer {
+        let mut state = serializer.serialize_struct("SerializeEvent", 2)?;
+        state.serialize_field("metadata", &SerializeMetadata(self.0.metadata()))?;
+        state.end()
+    }
+}
+
+/// A Serde visitor to pull `Attributes` data out of a serialized stream
+#[derive(Debug)]
+pub struct SerializeAttributes<'a>(pub (crate) Attributes<'a>);
+
+impl<'a> Serialize for SerializeAttributes<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: Serializer {
+        let mut state = serializer.serialize_struct("SerializeAttributes", 3)?;
+        state.serialize_field("metadata", &SerializeMetadata(self.0.metadata()))?;
+        if let Some(id) = self.0.parent() {
+            // TODO this probably isn't how we want it
```

Comment:
I think we should just serialize the result of calling `self.0.parent()` as an `Option` and then serialize `self.0.is_root()` as a bool.